src/mod_zone_difficulty_scripts.cpp

```diff
@@ -537,6 +537,72 @@ class mod_zone_difficulty_worldscript : public WorldScript
         sZoneDifficulty->IsDebugInfoEnabled = sConfigMgr->GetOption<bool>("ModZoneDifficulty.DebugInfo", false);
         sZoneDifficulty->LoadMapDifficultySettings();
     }
+
+    void OnAfterUpdateEncounterState(Map* map, EncounterCreditType type, uint32 creditEntry, Unit* source, Difficulty difficulty_fixed, DungeonEncounterList const* encounters, uint32 dungeonCompleted, bool updated)
+    {
+        LOG_ERROR("sql.sql", "Encounter completed");
+        // todo: Apparently this doesn't fire.
+    }
+};
+
+class mod_zone_difficulty_dungeonmaster : public CreatureScript
+{
+public:
+    mod_zone_difficulty_dungeonmaster() : CreatureScript("mod_zone_difficulty_dungeonmaster") { }
+
+    bool OnGossipSelect(Player* player, Creature* /*creature*/, uint32 /*sender*/, uint32 action) override
+    {
+        if (action == 100)
+        {
+            LOG_ERROR("sql.sql", "Try turn on");
+            bool CanTurnOn = true;
+            // Was a single encounter completed on normal mode?
+            if (sZoneDifficulty->HardmodeInstanceData[player->GetMap()->GetInstanceId()].CompletedEncounterOnNormal != true)
```

Comment:
```suggestion
            if (sZoneDifficulty->HardmodeInstanceData[player->GetMap()->GetInstanceId()] && sZoneDifficulty->HardmodeInstanceData[player->GetMap()->GetInstanceId()].CompletedEncounterOnNormal)
```

Reply:
or, even better, assign to a variable the result of `sZoneDifficulty->HardmodeInstanceData[player->GetMap()->GetInstanceId()]` and use it in `if (someVariable && someVariable.CompletedEncounterOnNormal)

Reply:
![image](https://user-images.githubusercontent.com/71938210/210392146-84eb5313-0192-4b3d-b599-a6a69ed54d99.png)
&& is not accepted for reasons unknown to me :(